dali/operators/image/mask/grid_mask.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/core/static_switch.h"
+#include "dali/operators/image/mask/grid_mask.h"
+#include "dali/kernels/mask/grid_mask_cpu.h"
+#include "dali/core/format.h"
+
+#define TYPES (uint8_t, int16_t, int32_t, float)
+
+namespace dali {
+
+DALI_SCHEMA(GridMask)
+    .DocStr(R"(Zeroes out pixels of an image in a grid-like fashion. The grid
+consists of squares repeating in x and y directions, with the same spacing in
+both directions. Can be rotated around the origin.)")
+    .NumInput(1)
+    .NumOutput(1)
+    .AddOptionalArg("tile", R"(The length of a single tile, which is equal to
+width of black squares plus the spacing between them.)",
+                    100, true)
+    .AddOptionalArg("ratio",
+                    "The ratio between black square width and tile width.",
+                    0.5f, true)
+    .AddOptionalArg("angle",
```

Comment:
The mentioned paper mentions sigma - the distances between the first intact unit and boundary of the image, maybe we should add it as well.

Reply:
I added a translation vector argument, for our purposes i think it's ok